core/node/core.go

```diff
@@ -75,6 +80,10 @@ func (s *syncDagService) Sync() error {
 	return s.syncFn()
 }
 
+func (s *syncDagService) Session(ctx context.Context) format.NodeGetter {
```

Comment:
This lets mdag.FetchGraph correctly setup a bitswap session. A better fix is in #7201 but that's not going to make it into the release.